hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/clientrm/RouterClientRMService.java

```diff
@@ -623,4 +648,22 @@ public void initUserPipelineMap(Configuration conf) {
         YarnConfiguration.DEFAULT_ROUTER_PIPELINE_CACHE_MAX_SIZE);
     this.userPipelineMap = Collections.synchronizedMap(new LRUCacheHashMap<>(maxCacheSize, true));
   }
+
+  private URL getRedirectURL() throws Exception {
+    Configuration conf = getConfig();
+    String webAppAddress = WebAppUtils.getWebAppBindURL(conf, YarnConfiguration.ROUTER_BIND_HOST,
+        WebAppUtils.getRouterWebAppURLWithoutScheme(conf));
+    String[] hostPort = StringUtils.split(webAppAddress, ':');
+    if (hostPort.length != 2) {
+      throw new YarnRuntimeException("Router can't get valid redirect proxy url");
+    }
+    String host;
+    if (StringUtils.isBlank(hostPort[0]) || hostPort[0].equals("0.0.0.0")) {
+      host = InetAddress.getLocalHost().getCanonicalHostName();
+    } else {
+      host = hostPort[0];
+    }
+    return new URL(YarnConfiguration.useHttps(this.getConfig()) ? "https" : "http", host,
+        Integer.parseInt(hostPort[1]), "");
```

Comment:
Extract the port.
```
String scheme = YarnConfiguration.useHttps(this.getConfig()) ? "https" : "http";
int port = Integer.parseInt(hostPort[1]), "")
```

Reply:
@goiri  Thanks for you review, I have fix it.